From 0be35dfac5965bd9a8a9be4e5c10cb60b1702d85 Mon Sep 17 00:00:00 2001
From: Marcus Lundblad <ml@dfupdate.se>
Date: Wed, 9 Aug 2023 21:43:31 +0200
Subject: [PATCH 1/2] sidebar: Add method to focus the start route entry

---
 src/sidebar.js | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/sidebar.js b/src/sidebar.js
index 57668691..6b1ae51b 100644
--- a/src/sidebar.js
+++ b/src/sidebar.js
@@ -75,6 +75,10 @@ export class Sidebar extends Gtk.Grid {
             Application.routingDelegator.transitRouter.enabled;
     }
 
+    focusStartEntry() {
+        this._entryList.get_row_at_index(0).child.entry.grab_focus();
+    }
+
     _initTransportationToggles(pedestrian, bike, car, transit) {
         let transport = RouteQuery.Transportation;
 
-- 
2.41.0

